src/lib.rs

```diff
@@ -0,0 +1,198 @@
+//!    various purposes, with zero to little additional code needed. All of these templates contain
+//!    runtimes that are highly configurable and are likely suitable for basic needs.
+//! 2. Customizing the client: To the contrary, some developers may want to customize the client
+//!    side software to achieve novel goals such as a new consensus engine, or a new database
+//!    backend. While Substrate's main configurability is in the runtime, the client is also highly
+//!    generic and can be customized to a great extent.
+//!
+//! ## Structure
+//!
+//! Substrate is a massive cargo workspace with hundreds of crates, therefore it is useful to know
+//! how to navigate its crates.
+//!
+//! In broad terms, it is divided into three categories:
+//!
+//! * `sc-*` (short for *substrate-client*) crates, located under `./client` folder. These are all
+//!   the client crates. Notable examples are crates such as [`sc-network`], various consensus
+//!   crates, [`sc-rpc-api`] and [`sc-client-db`], all of which are expected to reside in the client
+//!   side.
+//! * `sp` (short for *substrate-primitives*) crates, located under `./primitives` folder. These are
+//!   the traits that glue the client and runtime together, but are not opinionated about what
+//!   framework is using for building the runtime. Notable examples are [`sp-api`] and [`sp-io`],
+//!   which form the communication bridge between the client and runtime, as explained in
+//!   [`substrate_diagram`].
+//! * `pallet-*` and `frame-*` crates, located under `./frame` folder. These are the crates related
+//!   to FRAME. See [`frame_support`] for more information.
+//!
+//! ## Parachain?
+//!
+//! As noted above, Substrate is the main engine behind the Polkadot ecosystem. One of the ways
+//! through which Polkadot can be utilized is by building "parachains", blockchains that are
+//! connected to Polkadot's shared security.
+//!
+//! To build a parachain, one needs to use [`Cumulus`](https://github.com/paritytech/cumulus/), the
+//! library on top of Substrate, empowering any substrate-based chain to be a Polkadot parachain.
+//!
+//! ## Where To Go Next?
+//!
+//! Additional noteworthy crates within substrate:
+//!
+//! - RPC APIs of a Substrate node: [`sc-rpc-api`]
+//! - CLI Options of a Substrate node: [`sc-cli`]
+//! - All of the consensus related crates provided by Substrate:
+//! 	- [`sc-consensus-aura`]
+//! 	- [`sc-consensus-babe`]
+//! 	- [`sc-consensus-grandpa`]
+//! 	- [`sc-consensus-beefy`]
+//! 	- [`sc-consensus-manual-seal`]
+//! 	- [`sc-consensus-pow`]
+//!
+//! Additional noteworthy external resources:
+//!
+//! - [Substrate Developer Hub](https://substrate.dev)
+//! - [Parity Tech's Documentation Hub](https://paritytech.github.io/)
+//! - [Frontier: Substrate's Ethereum Compatibility Library](https://paritytech.github.io/frontier/)
+//! - [Polkadot Wiki](https://wiki.polkadot.network/en/)
+//!
```

Comment:
Maybe a section for interesting upstream deps like https://github.com/paritytech/parity-common and https://github.com/paritytech/parity-db?